tests/services/form_test.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+from httpx import AsyncClient
+from structlog.stdlib import BoundLogger
+
+from jupyterlabcontroller.config import Configuration
+from jupyterlabcontroller.services.form import FormManager
+from jupyterlabcontroller.services.prepuller import PrepullerManager
+
+
+@pytest.mark.asyncio
+async def test_generate_user_lab_form(
+    config: Configuration,
+    prepuller_manager: PrepullerManager,
+    logger: BoundLogger,
+    http_client: AsyncClient,
+) -> None:
+    lab_sizes = config.lab.sizes
+    fm: FormManager = FormManager(
+        prepuller_manager=prepuller_manager,
+        logger=logger,
+        http_client=http_client,
+        lab_sizes=lab_sizes,
+    )
+    r = await fm.generate_user_lab_form()
+    assert (
+        r.find(
+            '<option value="lighthouse.ceres/library/sketchbook:'
+            'recommended@sha256:5678">'
+        )
+        != -1
+    )
```

Comment:
Best practice here is to have the fully rendered form that you expect either as a data file in the tests directory (recommended) or a constant in this test and compare the entire returned form for equality.